clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerClientFactory.java

```diff
@@ -84,17 +91,31 @@ public synchronized <K, V> VeniceChangelogConsumer<K, V> getChangelogConsumer(St
             d2ControllerClient,
             globalChangelogClientConfig.getControllerRequestRetryCount());
       }
+      String consumerName = storeName + "-" + viewClass.getClass().getSimpleName();
       if (viewClass.equals(ChangeCaptureView.class.getCanonicalName())) {
         return new VeniceChangelogConsumerImpl(
             newStoreChangelogClientConfig,
-            consumer != null ? consumer : new KafkaConsumer<>(newStoreChangelogClientConfig.getConsumerProperties()));
+            consumer != null
+                ? consumer
+                : getConsumer(newStoreChangelogClientConfig.getConsumerProperties(), consumerName));
       }
       return new VeniceAfterImageConsumerImpl(
           newStoreChangelogClientConfig,
-          consumer != null ? consumer : new KafkaConsumer<>(newStoreChangelogClientConfig.getConsumerProperties()));
+          consumer != null
+              ? consumer
+              : getConsumer(newStoreChangelogClientConfig.getConsumerProperties(), consumerName));
     });
   }
 
+  private PubSubConsumerAdapter getConsumer(Properties consumerProps, String consumerName) {
+    KafkaPubSubMessageDeserializer kafkaPubSubMessageDeserializer = new KafkaPubSubMessageDeserializer(
```

Comment:
Will we need to change this interface at some point?  KafkaPubSubMessageDeserializer seems a concretion and the code here doesn't allow it to be swapped out.  Do we have a plan for this for later?

Reply:
Yeah, your suggestion reminds that putting this `KafkaPubSubMessageDeserializer` inside the `ApacheKafkaConsumerAdapterFactory`, as xinfra adapter probably need a different serializer, I think @sushantmane was doing sth about this deserialzer.